storage/src/helpers/block_locators.rs

```diff
@@ -76,6 +77,12 @@ impl<N: Network> FromBytes for BlockLocators<N> {
     fn read_le<R: Read>(mut reader: R) -> IoResult<Self> {
         let num_locators: u32 = FromBytes::read_le(&mut reader)?;
 
+        // Check that the number of block_locators is less than the total MAXIMUM_BLOCK_LOCATORS.
+        if num_locators > MAXIMUM_BLOCK_LOCATORS {
+            error!("The list of block locators is too long");
+            return Err(ErrorKind::Other.into());
```

Comment:
My only concern here is that the peer that sends too many block locators will get insta-kicked (i.e. without going through the `add_failure` mechanism) by the node, but since this is technically protocol violation and this deserialization is costly, it might be preferable to do that.